pkg/actions/lua/lakefs/catalogexport/delta_exporter.lua

```diff
@@ -0,0 +1,133 @@
+local lakefs = require("lakefs")
+local formats = require("formats")
+local pathlib = require("path")
+local json = require("encoding/json")
+local storage_utils = require("lakefs/catalogexport/storage_utils")
+local utils = require("lakefs/catalogexport/internal")
+
+--[[
+    action:
+        - repository_id
+        - commit_id
+
+    hook_args:
+        - table_paths: ["path/to/table1", "path/to/table2", ...]
+        - lakefs_key
+        - lakefs_secret
+        - region
+
+    storage_client:
+        - put_object: function(bucket, key, data)
+
+]]
+local function export_delta_log(action, hook_args, storage_client)
+    local repo = action.repository_id
+    local commit_id = action.commit_id
+    local table_paths = hook_args.table_paths
+    local lakefs_key = hook_args.lakefs_key
+    local lakefs_secret = hook_args.lakefs_secret
+    local region = hook_args.region
+
+    local ns = utils.get_storage_namespace(repo)
+    if ns == nil then
+        error("failed getting storage namespace for repo " .. repo)
+    end
+    local delta = formats.delta_client("s3", lakefs_key, lakefs_secret, region)
```

Comment:
Not fully in the context yet but I can't help not wonder already. 
Is there a reason to depend on passing lakeFS credentials here? 

Until now we worked with the policy that lua lakeFS client is using the context of the logged in user in the context during runtime to authenticate with lakeFS api.

Reply:
As mentioned F2F, this is due to the need to supply the delta-go package's client AWS key and secret to communicate with the S3 gateway.
On top of that, the administrator might configure a user that has permission to read the Delta table from its location such that every commit will export it even if the user that committed it doesn't have the right permissions.